tests/ui/methods.stderr

```diff
@@ -423,83 +423,109 @@ error: use of `unwrap_or` followed by a function call
 342 |     let _ = stringy.unwrap_or("".to_owned());
     |                     ^^^^^^^^^^^^^^^^^^^^^^^^ help: try this: `unwrap_or_else(|| "".to_owned())`
 
+error: use of `expect` followed by a function call
+   --> $DIR/methods.rs:365:26
+    |
+365 |     with_none_and_format.expect(&format!("Error {}: fake error", error_code));
+    |                          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ help: try this: `unwrap_or_else(|| panic!(&format!("Error {}: fake error", error_code)))`
```

Comment:
So... I' willing to accept the PR as it is, since the suggestion is sensible for any input other than `&format!(...)`.

In the `format` case, we should get rid of the macro call and the reference and just call the `panic!` macro.

We can either merge this PR and open an issue about it, or you can fix it in this PR. What do you prefer?

Reply:
I see what you mean, I'll have a go at fixing it in this PR.